js/core/component.js

```diff
@@ -532,173 +525,39 @@ var Component = Class.inherit({
-                cachedSetters[name] = coreDataUtils.compileSetter(name);
-            }
-
-            var path = name.split(/[.[]/);
-
-            cachedSetters[name](that._options, value, {
-                functionsAsIs: true,
-                merge: !that._getOptionsByReference()[name],
-                unwrapObservables: path.length > 1 && !!that._getOptionsByReference()[path[0]]
-            });
-        };
-
-        var setOption = function(that, name, value) {
-            var previousValue = getOptionValue(that._options, name, false);
-
-            if(that._optionValuesEqual(name, previousValue, value)) {
-                return;
-            }
-
-            if(that._initialized) {
-                that._optionChanging(name, previousValue, value);
-            }
-
-            setOptionValue(that, name, value);
-            that._notifyOptionChanged(name, value, previousValue);
-        };
-
-        return function(options, value) {
-            var that = this,
-                name = options;
-
-            if(arguments.length < 2 && typeUtils.type(name) !== "object") {
-                name = normalizeOptionName(that, name);
-                return getOptionValue(that._options, name);
-            }
-
-            if(typeof name === "string") {
-                options = {};
-                options[name] = value;
-            }
-
-            that.beginUpdate();
-
-            try {
-                var optionName;
-                for(optionName in options) {
-                    prepareOption(that, options, optionName, options[optionName]);
-                }
-                for(optionName in options) {
-                    setOption(that, optionName, options[optionName]);
-                }
-            } finally {
-                that.endUpdate();
-            }
-        };
-    })(),
+    option: function(options, value) {
+        if(arguments.length < 2 && typeUtils.type(options) !== "object") {
+            return this._optionManager.getValue(options);
+        }
 
-    _getOptionValue: function(name, context) {
-        var value = this.option(name);
+        this.beginUpdate();
 
-        if(isFunction(value)) {
-            return value.bind(context)();
+        try {
+            options = this._convertToObj(options, value);
+            this._optionManager.setValue(options);
+        } finally {
+            this.endUpdate();
         }
+    },
 
-        return value;
+    /**
+     * @name componentmethods.resetOption
+     * @publicName resetOption()
+     */
+    /**
+     * @name componentmethods.resetOption
+     * @publicName resetOption(optionName)
+     * @param1 optionName:string
+     */
+    resetOption: function(name) {
+        if(!name) {
+            return;
+        }
+        const defaultValue = this.initialOption(name);
+        this.beginUpdate();
+        const option = this._convertToObj(name, defaultValue);
+        this._optionManager.setValue(option);
```

Comment:
No need to use a variable

```suggestion
        this._optionManager.setValue(this._convertToObj(name, defaultValue));
```